### PR TITLE
mobile: refactor add device, bug fixes

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
- Fix Computer getting added to a phone slot
- Fix slot incrementing after addition
- improve copy
- unify add passkey / add device code
- free negative diff 🤷‍♂️ 

fixes some of the things in #615

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/keyRotation/AddKeySlotButton.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      return {
+        ...acc,
+        pendingKeyRotation: [...acc.pendingKeyRotation, pendingOp],
+      };
+    },
+  });
+
+  const didUserCancel =
+    message.includes("User cancelled") || // Android
+    message.includes("User canceled"); // iOS
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/keyRotation/AddKeySlotButton.tsx:83]: lol
we might want to factor this out into a function in common?

File: apps/daimo-mobile/src/view/screen/keyRotation/AddPasskeyScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const goBack = useExitBack();
+  const goHome = useExitToHome();
+
+  return (
+    <View style={ss.container.screen}>
+      <ScreenHeader title="Passkey Backup" onBack={goBack || goHome} />
+      <Spacer h={32} />
+      <View style={ss.container.padH16}>
+        <TextPara>
+          Back up your account by saving a secure passkey in {cloudName}.
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/keyRotation/AddPasskeyScreen.tsx:38]: maybe `in your password manager, for example {cloudName}.`

Saw a 1password user get confused by this ("but i'm not using GPM...?")

File: apps/daimo-mobile/src/view/screen/onboarding/UseExistingPage.tsx [hunk 3]
```diff
@@ -68,34 +71,37 @@ export function UseExistingPage({
...
+          {useExistingStatus === "error" ? (
             <TextLight>
               <EmojiToOcticon size={16} text={useExistingMessage} />
             </TextLight>
+          ) : (
+            <TextPara>
+              Add this {env(daimoChain).deviceType} to an existing account. Go
+              to Daimo {`>`} Settings {`>`} Add Device on your other device to
+              scan the QR code.
+            </TextPara>
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/onboarding/UseExistingPage.tsx:94]: great call

**Review by dcposch — APPROVED** [review]
LGTM after comments

File: apps/daimo-mobile/src/view/screen/keyRotation/AddPasskeyScreen.tsx (hunk 2 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/keyRotation/AddPasskeyScreen.tsx]: done

**Pull request merged**